site/docs/components/dialog.mdx

```diff
@@ -0,0 +1,613 @@
+---
+name: Dialog
+menu: Components
+route: /components/dialog
+---
+
+import { Playground } from "docz";
+import { StatusLabel, Button, Dialog, TextArea, TextInput, IconInfoCircle, IconQuestionCircle, IconAlertCircle, IconPlusCircle, IconTrash } from "hds-react";
+
+import ColorBox from "../../src/components/ColorBox";
+import LargeParagraph from "../../src/components/LargeParagraph";
+import Link from "../../src/components/Link";
+import Text from "../../src/components/Text";
+
+# Dialog
+
+<StatusLabel>New</StatusLabel>
+<StatusLabel type="alert" style={{marginLeft: 'var(--spacing-xs)'}}>Pre-release</StatusLabel>
+<StatusLabel type="success" style={{marginLeft: 'var(--spacing-xs)'}}>Accessible</StatusLabel>
+
+<LargeParagraph>
+    Dialogs initiate a conversation between the service and the user. They are used when an input or a confirmation is needed from the user or when important information needs to be conveyed.
+</LargeParagraph>
+
+## Principles
+- Dialogs capture the browser focus and the user is forced to react on the dialog.
+    - To emphasize this, a dark screen overlay must be used to cover the view content behind the dialog element.
+- **Dialogs are very intrusive pattern and they should only be used when the immediate actions or focus from the user are needed.**
+- Be careful when including a separate close ("x") icon in the dialog. If there are more than one action available in the dialog, it can be ambiguous for the user which action is triggered when the close icon is pressed.
+    - Generally it is a good practise to omit the close icon if the dialog has more than one action available.
+- If your dialog contains form elements, follow [HDS form](/patterns/forms) and [form validation patterns](/patterns/form-validation) similarly as you would in a regular forms.
+- As dialogs always contain buttons, pay close attention that the button labels describe the action it is going to trigger. You can read more information about this in the [HDS Button documentation page](/components/button).
+    - Following the same guidelines as the HDS Form pattern, dialog action buttons are placed at the left side of the dialog and the primary action will be the first one from the left.
```

Comment:
Should we add note here as well that on danger dialog's primary action is cancel so that the user doesn't accidentally destroy anything? It is mentioned on danger dialog but I guess it wouldn't harm to have it here as well?